nnpdf_data/nnpdf_data/new_commondata/LHCB_Z0_13TEV_2022/filter.py

```diff
@@ -0,0 +1,333 @@
+import yaml
+import numpy as np
+from collections import defaultdict, namedtuple
+
+# Loading tables
+with open('metadata.yaml', 'r') as file:
+        metadata = yaml.safe_load(file)
+
+# Initialise dictionaries for data and all
+# sources of uncertainties
+data = defaultdict(list)
+statcorr = defaultdict(list)
+effcorr = defaultdict(list)
+syserr = defaultdict(list)
+ndata_obs = defaultdict(list)
+observables = []
+
+# Loop over the implemented observables
+for observable in metadata['implemented_observables']:
+    # Append observable name and the number of points
+    observables.append(observable['observable_name'])
+    ndata_obs[observable['observable_name']] = observable['ndata']
+
+    # Append bin values
+    with open("rawdata/table_" + str(observable['tables'][0]) + ".yaml", 'r') as file:
+     data[observable['observable_name']] = yaml.safe_load(file)
+
+    # Append systematic uncertainties
+    with open("rawdata/table_" + str(observable['tables'][1]) + ".yaml", 'r') as file:
+     syserr[observable['observable_name']] = yaml.safe_load(file)
+
+    # Append statistical uncertainties
+    with open("rawdata/table_" + str(observable['tables'][2]) + ".yaml", 'r') as file:
+     statcorr[observable['observable_name']] = yaml.safe_load(file)
+
+    # Append efficiency uncertainties
+    with open("rawdata/table_" + str(observable['tables'][3]) + ".yaml", 'r') as file:
+     effcorr[observable['observable_name']] = yaml.safe_load(file)
+
+
+
+# Utility functions
+# ________________________________________________________
+def OuterDecomposition(matrix):
+    """ Compute the single value decomposition of the matrix A.
```

Comment:
```suggestion
    r""" Compute the single value decomposition of the matrix A.
```

Because you are using latex in this docstr, tell python it is a raw string (otherwise python will interpret all the `\` as escape sequences).

(not that it matters for the filters, nobody is going to inspect these docstrings from within python and be surprised, but it's just good practice to do so)

The same needs to be done in the other `filter.py` file.